test/integration/controller/core/workload_controller_test.go

```diff
@@ -469,5 +471,73 @@ var _ = ginkgo.Describe("Workload controller", ginkgo.Ordered, ginkgo.ContinueOn
 				}, maxExecTime, util.Interval).Should(gomega.Succeed())
 			})
 		})
+		ginkgo.It("should deactivate the workload when the time expires with multiple admissions", func() {
+			// due time rounding in conditions, the workload will stay admitted
+			// for a time between maxExecutionTime - 1s and maxExecutionTime
+			maxExecTime := 10 * time.Second
```

Comment:
```suggestion
			maxExecTime := 30 * time.Second
```

Should we increase this time since we can imagine the situations where each Eventually (`creating the workload and reserving its quota`, `waiting for the workload to be admitted, and for the time to change the second`, and `FinishEvictionForWorkloads `) takes the max time (5 sec).

In this situation, I guess that Line 515 will fail, and the case will be flake testing.